### PR TITLE
[do not merge] S2EF mappings

### PR DESCRIPTION
**Pull request by mshuaibii**
code review. General logic - will update for path specifics, etc.

### CODE REVIEW

File: experimental/s2ef_mapping.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+for system in old_system:
+    randomid = os.path.basename(system)[:-5]
+    old_images = ase.io.read(system, ":")
+    old_energies = [image.get_potential_energy() for image in old_iamges]
+
+    new_images = ase.io.read(f"{new_system_path}/{randomid}.traj", ":")
+    new_energies = [image.get_potential_energy() for image in new_iamges]
+
+    dict_mapping = {}
+    for old_idx, energy in enumerate(old_energies):
```
Comment (mshuaibii) [inline experimental/s2ef_mapping.py:17]: I was also thinking we can just `argsort` both new/old energies and then that will be a direct 1-1 mapping. Not sure if any edge cases there though.

Comment (mshuaibii) [reply experimental/s2ef_mapping.py:17]: actually leaving it as is would serve as an extra validation things are the same

File: experimental/s2ef_mapping.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import pickle
+import os
+import multiprocessing as mp
+import numpy as np
+
+
+def get_mapping(system):
+    randomid = os.path.basename(system)[:-7]
+    old_images = ase.io.read(system, ":")
+    old_energies = [image.get_potential_energy() for image in old_images]
```
Comment (sidgoyal78) [inline experimental/s2ef_mapping.py:13]: is looking at potential energy sufficient?
What if 2 frames have same potential energies?
Should we pair PE with some other aspect?

Comment (mshuaibii) [reply experimental/s2ef_mapping.py]: Switched to positions instead, those will be unique and safer than energies.

File: experimental/s2ef_mapping.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        new_images = ase.io.read(f"/home/jovyan/projects/ocp/data/trajs/train_02_01/{randomid}.traj", ":")
+        new_pos = [image.get_positions() for image in new_images]
+
+        dict_mapping = {}
+        # maps old frame idx -> new frame idx
+        for old_idx, pos in enumerate(old_pos):
+            for new_idx, pos_2 in enumerate(new_pos):
+                if (pos == pos_2).all():
+                    dict_mapping[old_idx] = new_idx
+                    break
```
Comment (sidgoyal78) [inline experimental/s2ef_mapping.py:25]: maybe we should also add an assert that all the keys are values are unique ?

**Review by sidgoyal78 — COMMENTED** [review]
Thanks for the PR, added one inline comment.

File: experimental/s2ef_mapping.py (hunk 3 above)
Comment (sidgoyal78) [reply experimental/s2ef_mapping.py:25]: Something like:
len(set(dict_mapping.keys())) == len(set(dict_mapping.values()))

File: experimental/s2ef_mapping.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+def get_mapping(randomid):
+    try:
+        old_images = read_old(randomid)
+        old_pos = [image.get_positions() for image in old_images]
+
+        new_images = read_new(randomid)
+        new_pos = [image.get_positions() for image in new_images]
+
+        visited = [False] * len(new_pos)
```
Comment (sidgoyal78) [inline experimental/s2ef_mapping.py:41]: yeah, thanks for updating this.

File: experimental/modify_frames.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    with open(
+        os.path.join("02_04_updated_ef_splits", os.path.basename(path)), "a"
+    ) as f:
+        for line in tqdm(open(path)):
+            try:
+                olddir, frame, total, randomid = line.split()[0].split(",")
+                newframe = str(mapping[randomid][int(frame)])
+                newdir = os.path.join(dirmapping[split], randomid+".traj")
+                new_line = ",".join([newdir,newframe,total, randomid])+"\n"
+                f.write(new_line)
```
Comment (sidgoyal78) [inline experimental/modify_frames.py:27]: yeah, this looks good.

**Review by sidgoyal78 — APPROVED** [review]
lgtm

**Pull request closed without merging**